azurelinuxagent/common/utils/restutil.py

```diff
@@ -468,7 +481,10 @@ def http_request(method,
                     # retry attempts
                     if _is_throttle_status(resp.status):
                         was_throttled = True
-                        max_retry = max(max_retry, THROTTLE_RETRIES)
+                        # Today, THROTTLE_RETRIES set to big number as opposite to slow down the retry attempts
```

Comment:
I don't follow this comment, could you please explain:
> Today, THROTTLE_RETRIES set to big number as opposite to slow down the retry attempts

Reply:
Today when any api return throttle error, we make 26 attempts for retying as opposed to back off and make fewer attempts with good delay. 

My fix will change this behavior for only telemetry calls and keeping same for non-telemetry calls as I wasn't sure why it was implemented that way. 

I'll improve the comment

Reply:
Thanks, makes sense